HW_9/HW_9.1_Class_Company.py

```diff
@@ -0,0 +1,375 @@
+        :param name: Company Name
+        :param country: Country where Company is origin from
+        :param headquarter: City where company's headquarter is located
+        :param founded: Year when company was founded
+        :param ceo: Name and Last Name of current company's CEO
+        :param employees: Total amount of employees in 2023
+        """
+        self.__name = name
+        self.__country = country
+        self.__headquarter = headquarter
+        self.__founded = founded
+        self.__ceo = ceo
+        self.__employees = employees
+
+    @property
+    def name(self):
+        """Company name"""
+        return self.__name
+
+    @name.setter
+    def name(self, new_name):
+        """
+        Set Company name
+        :param new_name: new name of the company
+        """
+        if not new_name.isalpha():
+            raise ValueError('Company name can contain only letters')
+        if len(new_name) > 25:
+            raise ValueError('Company length cannot be longer than 25 characters')
+        self.__name = new_name
+
+    @name.deleter
+    def name(self):
+        """Delete Company name"""
+        raise NotImplementedError('It is not allowed to delete Company name')
+
+    @property
+    def country(self):
+        """Country of registration"""
+        return self.__country
+
+    @country.setter
+    def country(self, new_country):
+        """
+        Set new country of registration
+        :param new_country: new country of registration
+        """
+        if new_country.lower() == 'russia':
+            raise ValueError('russia is terrorist organization. Only civilized countries are allowed.')
+        self.__country = new_country
+
+    @country.deleter
+    def country(self):
+        """ Delete country of registration"""
+        raise NotImplementedError('It is not allowed to delete attribute Country')
+
+    @property
+    def headquarter(self):
+        """Headquarter"""
+        return self.__headquarter
+
+    @headquarter.setter
+    def headquarter(self, new_headquarter):
+        """
+        Set new location of headquarter
+        :param new_headquarter: new location of headquarter
+        """
+        if new_headquarter == self.__headquarter:
+            raise ValueError('Headquarter is already set to the indicated location')
+        self.__headquarter = new_headquarter
+
+    @property
+    def founded(self):
+        """Year of company's foundation"""
+        return self.__founded
+
+    @founded.setter
+    def founded(self, new_founded):
+        """
+        Set new year of company's foundation
+        :param new_founded: new year value
+        """
+        if not isinstance(new_founded, int):
+            raise TypeError('Year foundation can only be an integer value')
+        if new_founded > self.CURRENT_YEAR:
+            raise ValueError('Year of foundation cannot be in the future')
+        self.__founded = new_founded
```

Comment:
I do not think that it is logically correct. How can we change the foundation year? Looks like this field required only getter

Reply:
I also thought about it, but my intention was to have solution in case of mistake. I am not sure how it's handled in real cases.